python/ray/data/_expression_evaluator.py

```diff
@@ -57,7 +58,16 @@ def _eval_expr_recursive(
     if isinstance(expr, ColumnExpr):
         return batch[expr.name]
     if isinstance(expr, LiteralExpr):
-        return expr.value
+        # Broadcast literal value to match batch size
+        if isinstance(batch, pd.DataFrame):
+            # For pandas, create a Series with the literal value repeated
+            return pd.Series(expr.value, index=batch.index)
+        elif isinstance(batch, pa.Table):
+            # For Arrow, create an Array with the literal value repeated
+            return pa.array([expr.value] * len(batch))
```

Comment:
Use `BlockAccessor.fill_column` (and abstract block-format specific nuances in it)